### DISCUSSION TITLE
Is there any way to generate a union schema during build time so that we can use the static schema to compose supergraph?

### DISCUSSION THREAD

**Discussion by icypig** [Netflix/dgs-framework#1038] [Q&A]
As I know dgs supplied a query named _service so that we can use rover introspect way to publish the subgraph schema, I'm wondering if there is any way that dgs can generate a union graphql schema so that we can use rover with static schema file to compose supergraph? 

Thanks a lot~

**Comment by srinivasankavitha** [comment]
What do you mean by a union graphql schema? You mean for the whole composed graph that the DGS is a part of? That would be provided by the federated gateway. The DGS has no knowledge of the entire graph by itself.
